### PR TITLE
Improve PTB results

### PR DESCRIPTION
**Pull request by yzhang87**
I would suggest changing the default LSTM forget_bias = 0. I don't think it is useful for speech and NLP task.

### CODE REVIEW

File: src/operator/cudnn_rnn-inl.h [hunk 1]
```diff
@@ -526,7 +526,7 @@ class CuDNNRNNOp : public Operator {
   cudnnRNNInputMode_t input_mode_;
   cudnnDropoutDescriptor_t dropout_desc_;
   Storage::Handle dropout_states_, reserve_space_;
-  uint64_t seed_ = 1337ull;
+  uint64_t seed_ = 17 + rand() % 4096;  // NOLINT(runtime/threadsafe_fn)
```
Comment (piiswrong) [inline src/operator/cudnn_rnn-inl.h]: does cudnn require a seed?

File: example/gluon/word_language_model/train.py [hunk 2]
```diff
@@ -125,19 +119,19 @@ def train():
             hidden = detach(hidden)
             with autograd.record():
                 output, hidden = model(data, hidden)
-                L = loss(output, target)
+                L = mx.nd.mean(loss(output, target))
```
Comment (piiswrong) [inline example/gluon/word_language_model/train.py]: we usually use sum to make multi gpu training easier. Does this affect accuracy?

**Comment by piiswrong** [comment]
Thanks!

I think bias is already 0 by default. Please check

File: src/operator/cudnn_rnn-inl.h (hunk 1 above)
Comment (yzhang87) [inline src/operator/cudnn_rnn-inl.h]: Yes, in dropoutDesc, it needs pass the seed in.

File: example/gluon/word_language_model/train.py (hunk 2 above)
Comment (yzhang87) [inline example/gluon/word_language_model/train.py]: Should be fine. Just want to make the parameter consistent with other package, e.g. pytorch. I can update it (need to rescale the clipping and learning rate).

**Comment by yzhang87** [comment]
@piiswrong I think it still set to 1.0 in the master. Do you want me change it?
`@register
class LSTMBias(Initializer):
    """Initialize all bias of an LSTMCell to 0.0 except for
    the forget gate whose bias is set to custom value.
    Parameters
    ----------
    forget_bias: float, default 1.0
        bias for the forget gate. Jozefowicz et al. 2015 recommends
        setting this to 1.0.
    """
    def __init__(self, **forget_bias=1.0**):
        super(LSTMBias, self).__init__(forget_bias=forget_bias)
        self.forget_bias = forget_bias
 `

File: example/gluon/word_language_model/train.py (hunk 2 above)
Comment (piiswrong) [inline example/gluon/word_language_model/train.py]: Let's use sum. Mean is good on single gpu. but on multi-gpu it's clumsy.

**Comment by piiswrong** [comment]
https://github.com/dmlc/mxnet/blob/master/python/mxnet/gluon/rnn/rnn_layer.py#L369

lstmbias is not used by default for gluon

**Comment by yzhang87** [comment]
@piiswrong removed the forget_bias in model.py and nd.mean in train.py. Also updated the README with new hyperparameters.

File: src/operator/cudnn_rnn-inl.h (hunk 1 above)
Comment (piiswrong) [inline src/operator/cudnn_rnn-inl.h]: I'll merge this in first.

Later we might consider a static atomic seed that gets incremented each time to be deterministic.

**Comment by piiswrong** [comment]
Thanks!

**Comment by piiswrong** [comment]
@szha Let's update the tutorial to include these changes.
